src/statmodels.jl

```diff
@@ -159,44 +159,70 @@ df_residual(obj::RegressionModel) = error("df_residual is not defined for $(type
 
 ## Nms are the coefficient names, corresponding to rows in the table
 type CoefTable
-    mat::Matrix
+    cols::Tuple
     colnms::Vector
     rownms::Vector
-    pvalcol::Integer
-    function CoefTable(mat::Matrix,colnms::Vector,rownms::Vector,pvalcol::Int=0)
-        nr,nc = size(mat)
-        0 <= pvalcol <= nc || error("pvalcol = $pvalcol should be in 0,...,$nc]")
+    function CoefTable(cols::Tuple,colnms::Vector,rownms::Vector)
+        nc = length(cols)
+        nrs = map(length,cols)
+        nr = max(nrs...)
         length(colnms) in [0,nc] || error("colnms should have length 0 or $nc")
         length(rownms) in [0,nr] || error("rownms should have length 0 or $nr")
-        new(mat,colnms,rownms,pvalcol)
+        all(unique(nrs) .== nr) || error("Elements of cols should have equal lengths")
+        new(cols,colnms,rownms)
+    end
+    function CoefTable(mat::Matrix,colnms::Vector,rownms::Vector,pvalcol::Int=0)
+        nr,nc = size(mat)
+        cols = Any[mat[:, i] for i in 1:nc]
+        if pvalcol != 0                         # format the p-values column
+            cols[pvalcol] = convert(Vector{PValue}, cols[pvalcol])
+        end
+        cols = ntuple((i)->cols[i],nc)
```

Comment:
`tuple(cols...)` does the same. But, really, making it a vector would be simpler and more consistent with other fields (since changing the latter would break DataFrames). People can do `Any[v1, v2...]` to disable concatenation in 0.4.